msdsn/conn_str.go

```diff
@@ -72,6 +74,26 @@ type Config struct {
 	ConnTimeout time.Duration // Use context for timeouts.
 	KeepAlive   time.Duration // Leave at default.
 	PacketSize  uint16
+
+	// Kerberos authentication fields
+	// Path to krb5.conf file that contains Kerberos configuration information
+	Krb5ConfFilePath string
```

Comment:
The configuration is better, but not what I requested.

There should be zero "paths" in this configuration struct; all features must be able to be accessed without writing to disk. As such, the three paths you introduce, cache, keytab, KRB5config, all need to be the content of the file. If the cache needs to look something up in real time (like caching), then it should provide an interface or struct to do so. Not an actual path.

Reply:
We are using gokrb package for implementing kerberos authentication which requires file path instead of file contents. Please refer to below link:
https://github.com/jcmturner/gokrb5/blob/663478bf457f1fc3275973bea5b7b787cd332015/credentials/ccache.go#L66
The keytab, KRB5Config and cache files can be located anywhere, hence we have to give user the ability to configure the path accordingly. 

In our use case we want the go-mssql package to be able to connect to mssql server which has Kerberos authentication enabled and kerberos requires these files whose contents cannot be passed through connection string. This is something that needs to be configured from application that is using go-mssql package.

Reply:
If the contents of a file may change over time, for example say a session ticket, then we will need to be able to call a function that will resolve the data. If the data will not change over time, then the static data can be provided as-is in the struct.

Independent of that, the DSN (string) may include file paths that read data from the file system and load data into the struct.

Here is one way it could work with the CCache: 
```
type msdsn.Config struct {
    ...
    KerbCCache []byte // File []byte can be converted to *CCache with *CCache.Unmarshal.
    KerbCCache *gokerb5.CCache
    KerbCCache func(ctx context.Context) []byte
    KerbCCache func(ctx context.Context) *gokerb5.CCache
}
```
I don't know which of these variants would necessarily be best. For instance, if someone wanted to use Vault without writing to the disk, then they could grab current ticket in Vault, then assign directly to the config.

Reply:
Have replaced the file path with structs as you requested.